### ISSUE TITLE
Publish OpenWhisk Swift handler library as package 

### ISSUE THREAD

**Issue by jthomas** [apache/openwhisk-runtime-swift#4]
The current Swift files for [implementing the OpenWhisk handler](https://github.com/apache/incubator-openwhisk-runtime-swift/tree/master/core/swift3Action/spm-build) are included as raw source files. This leads to a complex and manual process for building Swift binaries locally, which doesn't feel very "Swifty". 

```
docker run -it -v "$HOME:/owexec" openwhisk/swift3action bash
apt-get install -y zip
cp /owexec/hello.swift /swift3Action/spm-build/main.swift 
cat /swift3Action/epilogue.swift >> /swift3Action/spm-build/main.swift
echo '_run_main(mainFunction:main)' >> /swift3Action/spm-build/main.swift
/swift3Action/spm-build/swiftbuildandlink.sh
cd /swift3Action/spm-build
zip /owexec/hello.zip .build/release/Action
exit
```

Publishing this code as an [official Swift package](https://packagecatalog.com/) would massively improve the developer experience.  I've already experimented with doing this and have an unofficial package available here: 
https://packagecatalog.com/package/jthomas/OpenWhiskAction

### usage

Developers include the package in their manifest

```swift
let package = Package(
    name: "Action",
    dependencies: [
        .Package(url: "https://github.com/jthomas/OpenWhiskAction.git", majorVersion: 0)
    ]
)
```

... and then simply reference the library in their code.

```swift
import OpenWhiskAction

func hello(args: [String:Any]) -> [String:Any] {
    if let name = args["name"] as? String {
      return [ "greeting" : "Hello \(name)!" ]
    } else {
      return [ "greeting" : "Hello stranger!" ]
    }
}

OpenWhiskAction(main: hello)
```

Compiling a binary for the platform them becomes a single command.

```
docker run --rm -it -v $(pwd):/swift-package openwhisk/swift3action bash -e -c "cd /swift-package && swift build -v -c release"
```

This is a huge improvements over the current process.

### process

Creating this package no code changes, it's just bundling the existing handler files in the Swift package format.
https://github.com/jthomas/OpenWhiskAction

This package can also be used when dynamically compiling Swift actions rather than having the scripts also manually copy files about.

**Comment by csantanapr** [comment]
This looks good to me to me, we just need to adjust it for new Package.swift for swift 4.

I would do this for swift 4 and forward and leave 3.1.1 alone.

**Comment by csantanapr** [comment]
If make it part of the main swift4 runtime, then it should be cover by the test suite.

**Comment by jthomas** [comment]
Can this be re-opened? It hasn't been fixed by #1. @paulcastro

**Comment by dgrove-oss** [comment]
Closing as stale
